src/OpenLoco/src/Map/MapGenerator/MapGenerator.cpp

```diff
@@ -189,44 +189,106 @@ namespace OpenLoco::World::MapGenerator
         return ((randVal & 0xFF) * landObj->numVariations) >> 8;
     }
 
+    static void applySurfaceStyleToMarkedTiles(HeightMap& heightMap, uint8_t surfaceStyle, bool requireMark)
+    {
+        for (auto pos : World::getDrawableTileRange())
+        {
+            const bool tileIsMarked = heightMap.isMarkerSet({ pos.x, pos.y });
+            if ((requireMark && !tileIsMarked) || (!requireMark && tileIsMarked))
+                continue;
+
+            auto tile = TileManager::get(pos);
+            auto* surface = tile.surface();
+            if (surface == nullptr)
+                continue;
+
+            surface->setTerrain(surfaceStyle);
+            auto res = getRandomTerrainVariation(*surface);
+            if (res)
+                surface->setVar7(*res);
```

Comment:
can we give a name to var7 plz

Reply:
Ha, turns out there was already a named function! 😄